lib/dry/configurable/config_for_update.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require "dry/core/constants"
+
+module Dry
+  module Configurable
+    # @api private
+    class ConfigForUpdate
```

Comment:
Maybe `Config::Transient` would be better, it would look nicer and indicate it's private nature better as a nested class.

Reply:
Yeah, this is a good suggestion, thanks! I took the idea and ran with it a little.

`Transient`, while definitely a good descriptor for the lifecycle of the object, feel a little too generic to me — it's not clear from the name what the transient object is _used for_.

So I changed it to `Config::ToConfigure`, and changed the method returning this to be `Config#to_configure`. There's a nice symmetry there and hopefully this reinforces that this transient object is intended to live only over the lifecycle of the `#configure` method.